cpp/server/proxy.h

```diff
@@ -0,0 +1,57 @@
+#ifndef CERT_TRANS_SERVER_PROXY_H_
+#define CERT_TRANS_SERVER_PROXY_H_
+
+#include <functional>
+#include <vector>
+
+#include "base/macros.h"
+
+
+struct evhttp_request;
+
+namespace ct {
+class ClusterNodeState;
+}  // namespace ct
+
+namespace util {
+class Executor;
+class Status;
+class Task;
+}  // namespace util
+
+
+namespace cert_trans {
+namespace libevent {
+class Base;
+}  // namespace libevent
+
+class JsonOutput;
+class UrlFetcher;
+
+
+class Proxy {
+ public:
+  typedef std::function<std::vector<ct::ClusterNodeState>()>
+      GetFreshNodesFunction;
+  Proxy(libevent::Base* base, JsonOutput* output,
+        const GetFreshNodesFunction& get_fresh_nodes, UrlFetcher* fetcher,
+        util::Executor* executor);
+
+  virtual ~Proxy() = default;
+
+  virtual void ProxyRequest(evhttp_request* req) const;
```

Comment:
Yeah, it's to leave the door open for Mocks, I didn't add one in this PR though.